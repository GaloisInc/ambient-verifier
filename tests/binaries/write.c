#include <unistd.h>
#include <sys/syscall.h>

/**
 * This test verifies that the `write` syscall is connected. We don't currently
 * have a great way to verify that it works (the underlying machinery is tested
 * in crucible-symio and crucible-llvm), so we are mostly just testing that the
 * syscall number mapping is correct (i.e., the verifier should not error out
 * when it hits the syscall).
 */

char buf[100];

int main(void) {
  // Write to stdout
  syscall(SYS_write, 1, buf, 1);
}

void _start(void) {
  main();
}
