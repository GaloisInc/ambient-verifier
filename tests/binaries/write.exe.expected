successful: 0
failed: 2
wmTargets: []
fsRoot: tests/filesystems/read
